Patch up 2.5.2

JanneKiiskila opened the pull request:
To fix the build issues we are having with 2.5.1 release.

Commit:
Fix fcce build

Changed to use mbed-tools instead of mbed-cli.

(cherry picked from commit 762754da5df6d40a9d0ac5302f602076f1f7d130)
Signed-off-by: Janne Kiiskila <janne.kiiskila@pelion.com>

Commit:
mbed-fcce / do not deploy mbed-os.lib

Optimisation, no point in deploying Mbed OS in Linux based systems.

Commit:
CHANGELOG.md - 2.5.2

Add also CHANGELOG.md entry.

petedyerarm commented:
We could also take a copy of the fixed version of runc-opencontainers_git.bb into recipes-containers/runc

JanneKiiskila commented:
Good idea.

petedyerarm commented:
Also https://github.com/PelionIoT/meta-pelion-edge/recipes-misc/pelion-version/files/BUILDMMU.txt need updating

Commit:
Update version to 2.5.2

As per Pete's tip.

Commit:
Fix opencontainer/runc master/main build problem

By adding the recipe over here. This layer has priority 16,
so it precedes the broken recipe in meta-lmp-base (priority 9).

Commit:
conf/layer.conf - fix meta layer name (meta-edge)

Instead of referring to RPI, let's use our own meta layer name.

Commit:
CHANGELOG.md - add note about runc recipe

Add the note about runc recipe being pulled in.

Commit:
mbed-fcce - update Python version to 3.9

As 3.9 comes with the build now.
Amend list:
- Remove the double mbed-os.lib removal.
- Remove mbed-os.lib already earlier in do_configure
- Try straight rm, no ifs.
- Return back to if based rm.

JanneKiiskila committed:
Add openssl1.1.1o to recipes-connectivity

As the priority of this layer is higher than openembedded-core, this
gets pulled in rather than openssl1.1.1l, which has a rather unfortunate
bug in it (which got patched up later in openembedded-core with build target
reproducibility). This showed up as a rather strange problem with mbed-fcce,
where the build failed with a Python error similar to this:

|  File "/build/build-lmp/tmp-lmp/work/cortexa53-lmp-linux/mbed-fcce/4.11.1-r0/recipe-sysroot-native/usr/lib/python3.9/site-packages/pip/_vendor/urllib3/contrib/pyopenssl.py", line 46, in <module>
|    import OpenSSL.SSL
|  File "/build/build-lmp/tmp-lmp/work/cortexa53-lmp-linux/mbed-fcce/4.11.1-r0/recipe-sysroot-native/usr/lib/python3.9/site-packages/OpenSSL/__init__.py", line 8, in <module>
|    from OpenSSL import crypto, SSL
|  File "/build/build-lmp/tmp-lmp/work/cortexa53-lmp-linux/mbed-fcce/4.11.1-r0/recipe-sysroot-native/usr/lib/python3.9/site-packages/OpenSSL/crypto.py", line 1556, in <module>
|    class X509StoreFlags(object):
|  File "/build/build-lmp/tmp-lmp/work/cortexa53-lmp-linux/mbed-fcce/4.11.1-r0/recipe-sysroot-native/usr/lib/python3.9/site-packages/OpenSSL/crypto.py", line 1577, in X509StoreFlags
|    CB_ISSUER_CHECK = _lib.X509_V_FLAG_CB_ISSUER_CHECK
| AttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'
| WARNING: exit code 1 from a shell command.

For info on the patch (later then migrated into the openssl itself).

https://source.foundries.io/lmp-mirrors/openembedded-core.git/tree/meta/recipes-connectivity/openssl/openssl/reproducible.patch?h=hardknott

petedyerarm commented on CHANGELOG.md:
```diff
@@ -1,3 +1,8 @@
+
+- Fix compilation error related to `mbed-fcce`.
+- Pull in `opencontainer/runc` recipe to `meta-pelion-edge` to fix master to main rename build breakage.
+
```
Think we've changed more than this.

JanneKiiskila replied:
True, updated the CHANGELOG.md accordingly.

JanneKiiskila committed:
CHANGELOG.md - make it more complete

More detailed list of changes done.

petedyer11 reviewed (approved):
LGTM
